### DISCUSSION TITLE
Will udev monitor USB slave events?

### DISCUSSION THREAD

**Discussion by jimmbelll** [pyudev/pyudev#514] [Q&A]
I'm using a raspberry pi zero in USB slave mode to a host machine (HID device).  I'd like notifications when the host is connected or disconnected. Are USB slave notifications in the domain of what libudev handles? Monitoring doesn't seem to be returning them, though I may not be configured correctly.

```
import pyudev
c = pyudev.Context()
m = pyudev.Monitor.from_netlink(c)
m.filter_by('usb')
```

**Comment by mulkieran** [comment]
I'm not sure about the specifics. You should consider using ```udevadm monitor``` tool to see if it picks up the events. If it does and pyudev doesn't, then it's probably worth digging into the details.

**Reply by mulkieran** [reply]
USB notifications are definitely something that pyudev should be able to handle. It's the additional specifics in your question that I'm not sure how to answer.

**Comment by jimmbelll** [comment]
`sudo udevadm monitor`  shows nothing. I'm combing through its command-line parameters but not seeing anything that might help here.  So maybe USB slave here is outside of libudev's domain?

**Reply by mulkieran** [reply]
I'm afraid that is likely the case given what you've told me. pyudev doesn't keep up with enhancements in the underlying library, my role is just caretaker, really. But if you udevadm monitor is not picking the signals up either I would say that pyudev is very unlikely to.

You are confident that some udev add or remove signal is being sent on these changes? Because ultimately that depends on the implementation of the udev rules, which can be buggy.
